proto/internal/temporal/server/api/taskqueue/v1/message.proto

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.taskqueue.v1;
+
+option go_package = "go.temporal.io/server/api/taskqueue/v1;taskqueue";
+
+import "google/protobuf/empty.proto";
+
+// TaskVersionDirective controls how matching should direct a task.
+message TaskVersionDirective {
+    // Default value (if directive is not present):
+    // "unversioned": use the unversioned task queue, even if the task queue has
+    // versioning data.
+    oneof directive {
+        // If use_default is present, the task should be assigned the default
+        // version for the task queue. This will typically be set for the first
+        // workflow task in a workflow.
+        google.protobuf.Empty use_default = 1;
```

Comment:
Oh this is better than the bools I've used in other places.

Reply:
Looks like the JSON representation is the empty object, that sucks in JS land but doesn't apply in the server.

Reply:
Empty is literally just a pre-declared and shared `message Empty { }`

I wrote it as bool at first and then thought Spencer would complain that it allows a nonsense state: the oneof being set to this branch but it being false :grinning: 

The generated code for Empty is longer than for bool, sadly